app/models/vacancy.rb

```diff
@@ -40,9 +40,12 @@ class Vacancy < ApplicationRecord
       convert_date_to_unix_time(self.expires_on)
     end
 
+    attribute :expiry_time do
+      convert_time_to_unix_time(self.expiry_time)
```

Comment:
You don't need a separate method.  Just do:

```
time.nil? ? time : time.to_i
```

Reply:
How about:

```ruby
self.expiry_time.nil? ? self.expiry_time : convert_date_to_unix_time(self.expires_on)
```

- accounting for the case where we don't know `expiry_time` but we do have `expires_on` (date).